src/lib/detector.ts

```diff
@@ -54,3 +56,32 @@ export const supportsInterface = async (
     throw new Error(`Error checking the interface: ${error}`);
   }
 };
+
+function mapPermission(permission: string): string {
+  return LSP6_DEFAULT_PERMISSIONS[permission] || permission;
+}
+
+export const checkPermissions = (
+  requiredPermissions: string[] | string,
+  grantedPermissions: string,
+): boolean => {
+  // Convert requiredPermissions to an array if it's a single string
+  const requiredPermissionArray: string[] = Array.isArray(requiredPermissions)
+    ? requiredPermissions
+    : [requiredPermissions];
+
+  // Map the literal permissions to their bytes32 representation
+  const mappedPermissionArray: string[] =
+    requiredPermissionArray.map(mapPermission);
```

Comment:
What if the user passes a string which is not a permission? Will it handle it? Might be useful to add tests for this

Reply:
Added error handling and tests for when it's not a valid bytes32 hex string or known permission.